client/src/components/settings/toggle-radio-setting.tsx

```diff
@@ -50,17 +50,24 @@ export default function ToggleRadioSetting({
               name={flagName}
               value='1'
             />
+            <span className='custom-circle'>
+              {flag ? <span className='selected-inner-circle' /> : null}
+            </span>
             <span>{restProps.onLabel}</span>
           </label>
-          <label htmlFor={secondRadioId} data-checked={!flag}>
+
+          <label htmlFor={secondRadioId}>
             <input
               id={secondRadioId}
-              type='radio'
               {...(!flag && { defaultChecked: true })}
               {...(flag && { onChange: toggleFlag })}
               name={flagName}
+              type='radio'
               value='2'
-            />
+            />{' '}
+            <span className='custom-circle'>
+              {!flag ? <span className='selected-inner-circle' /> : null}
```

Comment:
```suggestion
              {!flag ? <span className='selected-inner-circle' /> : null}
```
We can delete `.selected-inner-circle` completely. Then we can use `.toggle-radio-group input:checked + .custom-circle:after` in CSS to add the inner circle.

Reply:
That works!! I implemented :checked pseudo class before as well, but removed it to maintain the consistency of old code. We could also remove `flag` and `toggleFlag` variable and function and add `checked` in 1st input html element, so its checked by default. It works fine!!

Reply:
No, we can't remove those properties on the `input`s. `defaultChecked` is used to make sure the 
correct radio button is checked when the page loads. `onChange` is what allows the user to change the setting (it calls `toggleFlag`, which you also got rid of). Getting rid of those two properties pretty much breaks the toggle. You need revert all of these changes. The only fixes you should be making are to the styling of the radio buttons, not their functionality.

Reply:
Ohh, it has been used in other files. Got it!!